tests/unit/test_column_noise.py

```diff
@@ -131,9 +139,100 @@ def test_miswrite_ages():
+                        "row_noise_level": 0.4,
+                        "token_noise_level": 0.5,
+                    },
+                },
+            },
+        }
+    )
+    config = config["decennial_census"]["street_number"]["numeric_miswriting"]
+    p_row_noise = config.row_noise_level
+    p_token_noise = config.token_noise_level
+    data = string_series
+    noised_data = _validate_seed_and_noise_data(
+        noise_type=NOISE_TYPES.NUMERIC_MISWRITING, column=data, config=config
+    )
+
+    # Get masks for helper groups, each string in categorical string purpose is to mimic possible string types
+    empty_str = data.str.len() == 0
+    unit_number = data.str.len() == 2
+    id_number = data.str.len() == 4
+    alt_str = data.str.len() == 6
+    income = data.str.len() == 9
+    date_of_birth = data.str.len() == 10
+    ssn = data.str.len() == 11
+    expected_noise = p_row_noise * p_token_noise * 0.9
+
+    # CHeck empty strings havent changed
+    assert (noised_data[empty_str] == "").all()
+
+    for i in range(2):  # "1A"
+        if i == 0:
+            assert np.isclose(
+                expected_noise,
+                (data[unit_number].str[i] != noised_data[unit_number].str[i]).mean(),
+                rtol=0.02,
+            )
+        else:
+            assert (data[unit_number].str[i] == noised_data[unit_number].str[i]).all()
+
+    for i in range(4):  # "1234"
+        assert np.isclose(
+            expected_noise,
+            (data[id_number].str[i] != noised_data[id_number].str[i]).mean(),
+            rtol=0.02,
+        )
+
+    for i in range(6):  # "a1b2c3"
+        if i % 2 == 0:
+            assert (data[alt_str].str[i] == noised_data[alt_str].str[i]).all()
+        else:
+            assert np.isclose(
+                expected_noise,
+                (data[alt_str].str[i] != noised_data[alt_str].str[i]).mean(),
+                rtol=0.02,
+            )
+
+    for i in range(9):  # "100000.00"
+        if i == 6:
+            assert (data[income].str[i] == noised_data[income].str[i]).all()
+        else:
+            assert np.isclose(
+                expected_noise,
+                (data[income].str[i] != noised_data[income].str[i]).mean(),
+                rtol=0.02,
+            )
+
+    for i in range(10):  # "12/01/2020"
+        if i in [2, 5]:
+            assert (data[date_of_birth].str[i] == noised_data[date_of_birth].str[i]).all()
+        else:
+            assert np.isclose(
+                expected_noise,
+                (data[date_of_birth].str[i] != noised_data[date_of_birth].str[i]).mean(),
+                rtol=0.02,
+            )
+
+    for i in range(11):  # "123-45-6789"
+        if i in [3, 6]:
+            assert (data[ssn].str[i] == noised_data[ssn].str[i]).all()
+        else:
+            assert np.isclose(
+                expected_noise,
+                (data[ssn].str[i] != noised_data[ssn].str[i]).mean(),
+                rtol=0.03,
```

Comment:
This fails with rtol=0.02 but passes with 0.03.   Should I change all of them or is it worth investigating?

Reply:
I think it's fine. After you make the adjustment to token level noise try again with `rtol=0.02` since you'll get different randomness. If it only passes at 0.03 after that, it's fine.